src/sentry/api/serializers/models/alert_rule_activations.py

```diff
@@ -0,0 +1,16 @@
+from sentry.api.serializers import Serializer, register
+from sentry.incidents.models.alert_rule_activations import AlertRuleActivations
+
+
+@register(AlertRuleActivations)
+class AlertRuleActivationsSerializer(Serializer):
+    def serialize(self, obj, attrs, user, **kwargs):
```

Comment:
This must return a typeddict class like:
```Python
class AlertRuleActivationsResponse(TypedDict):
    id: str,
    alertRuleId: str,
    dateCreated: datetime,
    finishedAt: datetime,
    metric_value: int,
    querySubscriptionId: str,
    isComplete: bool
```

Also, is there a reason we're converting all the ids to strings?

Reply:
🤔 hmmm

you sure? this is the api serializer - not the DRF serializer? (response - vs ingest serializer)

I'm just following the pattern i'm seeing for other serializers 🤷

Reply:
sorry I mean it must by typed to return that typeddict class like:
def serialize(self, obj, attrs, user, **kwargs) -> AlertRuleActivationsResponse: